### PR TITLE
Refactor analysis

### PR DESCRIPTION
**Pull request by jacobthill**
@justinlittman take a look at this when you get a chance and let me know if you have thoughts on further improvements. I works for sakip-sabanci and auc. I can't test on qnl because it needs to run from a machine with a blacklisted IP. Once this is pushed to master, we can run it from the harvest server. bnf uses a different kind of API.

### CODE REVIEW

File: README.md
Comment (justinlittman) [inline README.md]: A more typical way of doing this markdown would be:
\```
python oai-harvest.py http://cdm21044.contentdm.oclc.org/oai/oai.php -m oai_dc
\```

File: README.md
Comment (justinlittman) [inline README.md:36]: Would recommend using virtualenv and specifying those requirements in `requirements.txt`. (I can help with this.)

File: harvest-refactor/oai-harvest.py
Comment (justinlittman) [inline harvest-refactor/oai-harvest.py]: dest is redundant. It will default to dest. Same in others below.

File: harvest-refactor/oai-harvest.py
Comment (justinlittman) [inline harvest-refactor/oai-harvest.py]: If you use `action='store_true'` then a user only needs to provide the flag in the commandline (no need to do `-s TRUE`).

File: harvest-refactor/oai-harvest.py
Comment (justinlittman) [inline harvest-refactor/oai-harvest.py:50]: For bonus points, I would place the commandline parsing down here and have it invoke main() with arguments.

File: harvest-refactor/oai-harvest.py
Comment (justinlittman) [inline harvest-refactor/oai-harvest.py]: This seems wrong: For each record, write all records.

**Review by justinlittman — CHANGES_REQUESTED** [review]
You don't want to add the .pyc files. Adding a python-specific .gitignore will avoid this.

Happy to pair to clarify any of this.

**Comment by jacobthill** [comment]
Thanks @justinlittman for the feedback. I incorporated all suggestions except I made the requirements.txt with conda instead of virtualenv. I'm not sure if it is compatible with virtualenv but anyone should be able to recreate the environment with anaconda. Let me know if you have other thoughts. I still plan on refactoring the sru-harvest but I haven't had a need to use it beyond bnf so I'll hold off for now. If everything looks good, go ahead and merge and I can test it on the server to harvest the qnl data.

File: harvest-refactor/helper.py
Comment (justinlittman) [inline harvest-refactor/helper.py]: Would be clearer to just use `d.get(key, [])` rather than this function.

File: harvest-refactor/helper.py
Comment (justinlittman) [inline harvest-refactor/helper.py]: Can use `for record_number, record in enumerate(records):` instead of separately keeping track of record number. Only thing to be aware of is enumerate is 0 based.

File: harvest-refactor/helper.py
Comment (justinlittman) [inline harvest-refactor/helper.py]: Just do `os.makedirs(os.path.dirname(directory), exists_ok=True)` and can get rid of checking if the path exists and the error handling below.

File: harvest-refactor/helper.py
Comment (justinlittman) [inline harvest-refactor/helper.py]: Unnecessary. `close()` is handled by the `with`.

**Review by justinlittman — APPROVED** [review]
I missed `helper.py` before so added some comments. Also, still need to delete `helper.pyc`. After consideration of those changes, I approve.

**Comment by jacobthill** [comment]
Ok @justinlittman I made those changes as well.

**Pull request merged**
